src/components/types/datePicker.ts

```diff
@@ -19,6 +19,8 @@ export interface DateFieldBase<T = DateTime> extends ValueBase<T | null>, InputB
      * @default The timezone of the `value` or `defaultValue` or `placeholderValue`, 'default' otherwise.
      */
     timeZone?: string;
+    /** Custom parser function for parsing pasted date strings. If not provided, the default parser will be used. */
+    dateFromStringParser?: (dateStr: string, format: string, timeZone?: string) => DateTime;
```

Comment:
Can you please rename it to `parseDateFromString`? This will make the code a little easier to read. And in the place of use, you can write `props.parseDateFromString`